File: backend/Entities/Models/Veiculo.cs
using System;
using System.Collections.Generic;
using Entities.Model2;

namespace Entities.Models;

public partial class Veiculo
{
    public Guid Id { get; set; }

    public string Placa { get; set; } = null!;

    public Guid? ProprietarioId { get; set; }

    public int? VieculoTipoId { get; set; }

    public string? Renavan { get; set; }

    public int? Ano { get; set; }

    public string? Cor { get; set; }

    public string? NumeroCrv { get; set; }

    public string? Chassi { get; set; }

    public string? Combustivel { get; set; }

    public string? MarcaModelo { get; set; }

    public decimal? CapacidadePeso { get; set; }

    public decimal? CapacidadeM3 { get; set; }

    public int? Eixos { get; set; }

    public DateTime DataCadastro { get; set; }

    public virtual ICollection<AppRouteUserVeiculo> AppRouteUserVeiculos { get; } = new List<AppRouteUserVeiculo>();

    public virtual Pessoa? Proprietario { get; set; }

    public virtual ICollection<Veiculo1> Veiculo1s { get; } = new List<Veiculo1>();

    public virtual VeiculoTipo? VieculoTipo { get; set; }
}

File: frontend-with-menus/src/app/Business/Models/veiculo.ts
import { Pessoa } from "./pessoa";
import { Veiculo1 } from "./veiculo1";
import { VeiculoTipo } from "./veiculoTipo";

export interface Veiculo  {
    id: string;
    placa: string;
    proprietarioId: string | null;
    vieculoTipoId: number | null;
    renavan: string | null;
    ano: number | null;
    cor: string | null;
    numeroCrv: string | null;
    chassi: string | null;
    combustivel: string | null;
    marcaModelo: string | null;
    capacidadePeso: number | null;
    capacidadeM3: number | null;
    eixos: number | null;
    dataCadastro: string;
    proprietario: Pessoa | null;
    veiculo1s: Veiculo1[];
    vieculoTipo: VeiculoTipo | null;
}